packages/db/src/core/update.ts

```diff
@@ -14,39 +14,47 @@ import { Table } from "./createTable";
  * // Increase the age of Alice
  * await update(userTable, { id: userId, age: 16 });
  */
-export async function update<T, P extends keyof T>(
+export async function update<T extends object, P extends keyof T>(
   table: Table<T, P>,
   diff: Diff<T, P>
 ): Promise<void> {
   const primaryKeyProperty = table[BlinkKey].options.primary;
-  const primaryKey = diff[primaryKeyProperty] as unknown as T[P];
+  const primaryKey = diff[primaryKeyProperty];
+
+  if (primaryKey === undefined || primaryKey === null) {
+    throw new Error(`"${primaryKey}" is an invalid primary key value.`);
+  }
+
   const item = table[BlinkKey].storage.primary.get(primaryKey);
 
   if (item === undefined || item === null) {
     throw new Error(`Item with primary key "${primaryKey}" not found.`);
   }
 
   const oldItem = clone(item);
-  for (let key in diff) {
-    if (key === primaryKeyProperty) {
+
+  let key: keyof T;
+  for (key in diff) {
+    const diffValue = diff[key];
+    if (key === primaryKeyProperty || diffValue === undefined || diffValue === null) {
```

Comment:
This is questionable however. What is the expected behaviour when a value in diff is null or undefined?